code/modules/economy/holopay.dm

```diff
@@ -29,76 +24,74 @@
 	/// The brand icon chosen by the user
 	var/shop_logo = "donate"
 
-/obj/machinery/holopay/New(turf/location, mob/living/user, obj/item/card/id/card, datum/bank_account/account)
-	loc = location
-	owner = user
-	linked_card = card
-	linked_account = account
-	desc = "Pays directly into [owner]'s bank account."
+/**
+ * Links the source card to the holopay. Begins checking if its in range.
+ */
+/obj/effect/overlay/holopay/proc/assign_card(turf/target, obj/item/card/id/card)
+	loc = target
+	card_ref = WEAKREF(card)
+	desc = "Pays directly into [card.registered_account.account_holder]'s bank account."
 	add_atom_colour("#77abff", FIXED_COLOUR_PRIORITY)
 	set_light(2)
 	visible_message(span_notice("A holographic pay stand appears."))
 	/// Start checking if the owner has left or died
-	RegisterSignal(owner, list(COMSIG_MOVABLE_MOVED, COMSIG_LIVING_STATUS_UNCONSCIOUS), .proc/check_operation)
-	return ..()
+	RegisterSignal(card, COMSIG_MOVABLE_MOVED, .proc/check_operation)
+	return TRUE
 
 /**
- * A periodic check to see if the owner has left the area, died, or is afk.
- * Deletes the holopay if any condition is true.
+ * A periodic check to see if the projecting card is nearby.
+ * Deletes the holopay if true.
  */
-/obj/machinery/holopay/proc/check_operation()
+/obj/effect/overlay/holopay/proc/check_operation()
 	SIGNAL_HANDLER
-	if(!owner || !owner.client || !isliving(owner) || !IN_GIVEN_RANGE(src, owner, MAX_HOLO_RANGE))
-		QDEL_NULL(linked_card?.my_store)
+	var/obj/item/card/id/linked_card = card_ref.resolve()
+	if(QDELETED(linked_card) || !IN_GIVEN_RANGE(src, linked_card, max_holo_range))
+		playsound(loc, "sound/effects/empulse.ogg", 40, TRUE)
+		visible_message(span_notice("The pay stand vanishes."))
 		qdel(src)
 
-/obj/machinery/holopay/Destroy()
-	playsound(loc, "sound/effects/empulse.ogg", 40, TRUE)
-	visible_message(span_notice("The pay stand vanishes."))
-	return ..()
```

Comment:
I think what you want to do is move the sound and visual effect out of here, and into a new proc, maybe call it dissipate.
dissipate will do these two things, then call qdel on src

Then anywhere you want to cause this effect, use dissipate